File: algorithm/go/firstround/344.go
package firstround

func reverseString(s []byte) {
	if s == nil || len(s) < 2 {
		return
	}

	for left, right := 0, len(s)-1; left < right; left, right = left+1, right-1 {
		s[left], s[right] = s[right], s[left]
	}
	return
}

File: algorithm/cpp/firstround/344.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

class Solution
{
public:
    void reverseString(vector<char>& s);
};

void Solution::reverseString(vector<char>& s)
{
    int n = s.size();
    if(n<2) return ;

    int left=0, right=n-1;
    while (left<right)
    {
        swap(s[left], s[right]);
        left++;
        right--;
    }
    return ;
}

int main()
{
    return 0;
}